lib/widgets/code_block.dart

```diff
@@ -1,267 +1,549 @@
+      sx: TextStyle(color: const HSLColor.fromAHSL(1, 120, 1, 0.25).toColor()),
+
+      // .sr { color: hsl(189deg 54% 49%); }
+      sr: TextStyle(color: const HSLColor.fromAHSL(1, 189, 0.54, 0.49).toColor()),
+
+      // .s1 { color: hsl(86deg 57% 40%); }
+      s1: TextStyle(color: const HSLColor.fromAHSL(1, 86, 0.57, 0.40).toColor()),
+
+      // .ss { color: hsl(241deg 68% 28%); }
+      ss: TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor()),
+
+      // .bp { color: hsl(120deg 100% 25%); }
+      bp: TextStyle(color: const HSLColor.fromAHSL(1, 120, 1, 0.25).toColor()),
+
+      // .vc { color: hsl(241deg 68% 28%); }
+      vc: TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor()),
+
+      // .vg { color: hsl(241deg 68% 28%); }
+      vg: TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor()),
+
+      // .vi { color: hsl(241deg 68% 28%); }
+      vi: TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor()),
+
+      // .il { color: hsl(0deg 0% 40%); }
+      il: TextStyle(color: const HSLColor.fromAHSL(1, 0, 0, 0.40).toColor())
+    );
+  }
+
+  CodeBlockTextStyles._({
+    required this.plain,
+    required TextStyle hll,
+    required TextStyle c,
+    required TextStyle err,
+    required TextStyle k,
+    required TextStyle o,
+    required TextStyle cm,
+    required TextStyle cp,
+    required TextStyle c1,
+    required TextStyle cs,
+    required TextStyle gd,
+    required TextStyle ge,
+    required TextStyle gr,
+    required TextStyle gh,
+    required TextStyle gi,
+    required TextStyle go,
+    required TextStyle gp,
+    required TextStyle gs,
+    required TextStyle gu,
+    required TextStyle gt,
+    required TextStyle kc,
+    required TextStyle kd,
+    required TextStyle kn,
+    required TextStyle kp,
+    required TextStyle kr,
+    required TextStyle kt,
+    required TextStyle m,
+    required TextStyle s,
+    required TextStyle na,
+    required TextStyle nb,
+    required TextStyle nc,
+    required TextStyle no,
+    required TextStyle nd,
+    required TextStyle ni,
+    required TextStyle ne,
+    required TextStyle nf,
+    required TextStyle nl,
+    required TextStyle nn,
+    required TextStyle nt,
+    required TextStyle nv,
+    required TextStyle nx,
+    required TextStyle ow,
+    required TextStyle w,
+    required TextStyle mf,
+    required TextStyle mh,
+    required TextStyle mi,
+    required TextStyle mo,
+    required TextStyle sb,
+    required TextStyle sc,
+    required TextStyle sd,
+    required TextStyle s2,
+    required TextStyle se,
+    required TextStyle sh,
+    required TextStyle si,
+    required TextStyle sx,
+    required TextStyle sr,
+    required TextStyle s1,
+    required TextStyle ss,
+    required TextStyle bp,
+    required TextStyle vc,
+    required TextStyle vg,
+    required TextStyle vi,
+    required TextStyle il,
+  }) :
+    _hll = hll,
+    _c = c,
```

Comment:
Oof, lotta boilerplate.

But I think this is the structure we need in order to support dark mode (and particularly to support nicely lerping in and out of it), so that's just how it'll be. Definitely glad we have this off in its own file.